packages/aws-cdk/lib/api/logs/find-cloudwatch-logs.ts

```diff
@@ -84,38 +74,68 @@ function isReferencedFromIgnoredResource(
   logGroupResource: CloudFormation.StackResourceSummary,
   evaluateCfnTemplate: EvaluateCloudFormationTemplate,
 ): boolean {
-  let foundReference = false;
   const resourcesReferencingLogGroup = evaluateCfnTemplate.findReferencesTo(logGroupResource.LogicalResourceId);
-  for (const reference of resourcesReferencingLogGroup) {
-    if (IGNORE_LOGS_RESOURCE_TYPES.includes(reference.Type)) {
-      foundReference = true;
-    }
-  }
-  return foundReference;
+  return resourcesReferencingLogGroup.some(reference => {
+    return IGNORE_LOGS_RESOURCE_TYPES.includes(reference.Type);
+  });
 }
 
+type CloudWatchLogsResolver = (
+  resource: CloudFormation.StackResourceSummary,
+  evaluateCfnTemplate: EvaluateCloudFormationTemplate
+) => string | undefined;
+
+const cloudWatchLogsResolvers: Record<string, CloudWatchLogsResolver> = {
+  'AWS::Logs::LogGroup': (resource, evaluateCfnTemplate) => {
+    if (isReferencedFromIgnoredResource(resource, evaluateCfnTemplate)) {
+      return undefined;
+    }
+    return resource.PhysicalResourceId?.toString();
+  },
+
+  // Resource types that will create a CloudWatch log group with a specific name if one is not provided.
+  // The keys are CFN resource types, and the values are the name of the physical name property of that resource
+  // and the service name that is used in the automatically created CloudWatch log group.
+  'AWS::Lambda::Function': (resource, evaluateCfnTemplate) => {
```

Comment:
Lambda is not the only service that creates custom `logGroupName`s. We should note that similar functionality may be needed for several other services (rds, neptune, docdb, etc). However, Lambda Function was the only resource in `RESOURCE_TYPES_WITH_IMPLICIT_LOGS`, so we definitely do not need to add that functionality.

Reply:
I though about that while making this change but as the only reference I had was the Lambda one I thought in not adding more stuff than the things that already exists, but I can definitely work on that extras on upcoming PRs now that I got familiarity with the codebase.